### ISSUE TITLE
No beacons sent on Fairphone 2 (Fairphone OS 19.11.2 / Android 7.1.2)

### ISSUE THREAD

**Issue by gollub** [corona-warn-app/cwa-app-android#780]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

## Describe the bug

The Fairphone 2, currently running Android 7.1.2 with access to GMS, is not sending any beacon / exposure notification.

(Note: there are two firmware flavors, Fairphone Open OS and Fairphone OS. This is about the later one, which has GMS access)

This was noticed while playing with https://github.com/marq24/UUID0xFD6FTracer (@marq24 kudos!)

There is absolutely no indication to the user that the CWA App is not fully functional.

This behavior could be verified on two Fairphone 2 devices running the same Android / firmware. Also verified with other Bluetooth Beacon scanners that the Fairphone 2 OS is is never  sending a beacon. Did also a functional cross check with Samsung device.

Followed also the instructions from https://github.com/corona-warn-app/cwa-app-android/issues/674#issuecomment-646949156 to verify that the Fairphone 2 has no "Peripheral mode support" (and the Samsung has).

From the adb logs of the Fairphone 2:

`06-28 17:06:54.186  3370 22809 I ExposureNotification: Utils#isSupported enabled=true, isDeviceSupported=true, isBluetoothSupported=true, BluetoothAdapter.isMultipleAdvertisementSupported=false [CONTEXT service_id=236 ]`

## Expected behaviour

Give the users warning/notice/hint that this device is not able to send beacons, notify other devices.

Or even better would be to get exposure notification working. But unclear if that's possible. Will reach out to Fairphone team in parallel to clarify if they are aware of that issue.

## Steps to reproduce the issue

1. Buy a smartphone certified with the Blaue Engel by the Umweltbundesamt ;)
2. Install the CWA App as recommended by the RKI / Bundesministerium für Gesundheit ;)
3. Follow the instruction of the CWA App and grant the App access for "Use location" and enable Bluetooth

... the users get the impression that they actively participating, since no warnings or notice is given that the device is actually not capable sending beacons.

## Technical details

- Mobile device: Fairphone 2 / Fairphone OS 19.11.2
- Android version: Android 7.1.2

## Possible Fix

Detect Android devices with no "Peripheral mode support" and give user-friendly details that the CWA-App is not fully functional.

## Additional context

Fairphone 2 OS seems to get an upgrade to Android 9 in the near future: https://twitter.com/Fairphone/status/1272805847769460736
IIUC there are at least two more years spare-parts available: https://www.heise.de/newsticker/meldung/Fairphone-3-das-moeglichst-nachhaltig-Smartphone-4505848.html

**Comment by gollub** [comment]
Notified Fairphone team via support ticket.

**Comment by tkowark** [comment]
Thanks for providing this information. We'll discuss how to best address all the phones that don't support peripheral mode in the team and get back to you then.

Best regards,
TK
Corona-Warn-App Open Source Team

**Comment by cschweers** [comment]
according to iFixit the ([Fairphone 2](https://de.ifixit.com/Teardown/Fairphone+2+Teardown/52523#s115301)) uses Qualcomms **WCN3680B Wi-Fi/Bluetooth Combo**. And I fear it shares the same issue as the [Asus Nexus 7 (2013)](https://github.com/corona-warn-app/cwa-app-android/issues/674). 
To the best of my knowledge, the hardware is either not capable of running a proper Bluetooth low energy 'peripheral mode' or Qualcomm's Linux/Android driver does not support it.

**Comment by gollub** [comment]
This seems to indicate that the hardware is supporting it (different Firmware used):
https://forum.fairphone.com/t/corona-warn-app-zum-herunterladen/62003/74

**Comment by cschweers** [comment]
@gollub As far as I understand the thread you linked here, only one writes that he was able to activate the "peripheral mode". 
All others state that it does not work.
And I have never seen this combination of Bluetooth Low Energy Flags in nRF Connect before. I am sure that this is a consequence of the previous firmware hacking.

**Comment by tkowark** [comment]
To channel discussion about unsupported phones to a single issue, we will close this one here and refer to #688 for the time being.
